src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/ChatResponseExtensions.cs

```diff
@@ -123,30 +123,114 @@ static async Task AddMessagesAsync(
             list.AddMessages(await updates.ToChatResponseAsync(cancellationToken).ConfigureAwait(false));
     }
 
-    /// <summary>Combines <see cref="ChatResponseUpdate"/> instances into a single <see cref="ChatResponse"/>.</summary>
-    /// <param name="updates">The updates to be combined.</param>
-    /// <returns>The combined <see cref="ChatResponse"/>.</returns>
+    /// <summary>Applies a <see cref="ChatResponseUpdate"/> to an existing <see cref="ChatResponse"/>.</summary>
+    /// <param name="response">The response to which the update should be applied.</param>
+    /// <param name="update">The update to apply to the response.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="response"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="update"/> is <see langword="null"/>.</exception>
+    /// <remarks>
+    /// This method modifies the existing <paramref name="response"/> by incorporating the content and metadata
+    /// from the <paramref name="update"/>. This includes using <see cref="ChatResponseUpdate.MessageId"/> to determine
+    /// message boundaries, as well as coalescing contiguous <see cref="AIContent"/> items where applicable, e.g. multiple
+    /// <see cref="TextContent"/> instances in a row may be combined into a single <see cref="TextContent"/>.
+    /// </remarks>
+    [Experimental("MEAI0001")]
+    public static void ApplyUpdate(this ChatResponse response, ChatResponseUpdate update)
```

Comment:
Why do these ApplyUpdate(s) methods need to be public? I'm not seeing anything in this PR that necessitates that.

Reply:
I'm unclear why that's needed. If I'm consuming the streaming updates, rendering a partial update as I receive it, why do I need the partial coalescing behavior? I would assume I would just, e.g. update my UI with the new partial image as I receive it.

Reply:
I opened https://github.com/dotnet/extensions/issues/7017 to track this idea.